src/surface.rs

```diff
@@ -0,0 +1,53 @@
+//! Utility functions for creating dpi aware wayland surfaces.
+
+use std::sync::{Arc, Mutex};
+use wayland_client::Proxy;
+use wayland_client::protocol::wl_surface;
+use wayland_client::protocol::wl_compositor::{RequestsTrait as CompositorRequest};
+use crate::env::Environment;
+
+/// Creates a WlSurface from an Environment.
+/// Takes a callback for notification of dpi changes.
```

Comment:
Please add a newline between the title of the description and its body. Also please expand the documentation to detail how the dpi value is computed (taking the max of the dpi factors of all outputs this surface is displayed on).